python/tank/util/authentication.py

```diff
@@ -255,6 +257,53 @@ def logout():
         return False
 
 
+g_shotgun_current_user_cache = "unknown"
+
+
+def get_current_user(tk):
+    """
+    Retrieves the current user as a dictionary of metadata values.
+    Returns None if the user is not found in shotgun.
+
+    Returns the following fields:
```

Comment:
I know you are shuffling some existing code around, but would be nice if we could give this method a `returns` docstring at the same time!